shared/src/PayloadCollection.cs

```diff
@@ -0,0 +1,154 @@
+        /// </remarks>
+        /// <returns>The specified property.</returns>
+        public virtual object this[string key]
+        {
+            get => Collection[key];
+            set => AddOrUpdate(key, value);
+        }
+
+        /// <summary>
+        /// Adds the key-value pair to the collection.
+        /// </summary>
+        /// <inheritdoc cref="AddOrUpdate(string, object)" path="/param['key']"/>
+        /// <inheritdoc cref="AddOrUpdate(string, object)" path="/param['value']"/>
+        /// <inheritdoc cref="AddOrUpdate(string, object)" path="/exception"/>
+        /// <exception cref="ArgumentException">An element with the same key already exists in the collection.</exception>
+        public virtual void Add(string key, object value)
+        {
+            Collection.Add(key, value);
+        }
+
+        /// <summary>
+        /// Adds or updates the key-value pair to the collection.
+        /// </summary>
+        /// <param name="key">The name of the telemetry.</param>
+        /// <param name="value">The value of the telemetry.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <c>null</c> </exception>
+        public virtual void AddOrUpdate(string key, object value)
+        {
+            Collection[key] = value;
+        }
+
+        /// <summary>
+        /// Gets the collection as a byte array
+        /// </summary>
+        /// <remarks>
+        /// This will get the fully encoded serialized string using both <see cref="PayloadSerializer.SerializeToString(object)"/>
+        /// and <see cref="PayloadEncoder.EncodeStringToByteArray(string)"/> methods implemented in the <see cref="PayloadConvention"/>.
+        /// </remarks>
+        /// <returns>A fully encoded serialized string.</returns>
+        public virtual byte[] GetPayloadObjectBytes()
+        {
+            return Convention.GetObjectBytes(Collection);
+        }
+
+        /// <summary>
+        /// Determines whether the specified property is present.
+        /// </summary>
+        /// <param name="key">The key in the collection to locate.</param>
+        /// <returns><c>true</c> if the specified property is present; otherwise, <c>false</c>.</returns>
+        public bool Contains(string key)
+        {
+            return Collection.ContainsKey(key);
+        }
+
+        /// <summary>
+        /// Gets the value of the object from the collection.
+        /// </summary>
+        /// <remarks>
+        /// This class is used for both sending and receiving properties for the device.
+        /// </remarks>
+        /// <typeparam name="T">The type to cast the object to.</typeparam>
+        /// <param name="key">The key of the property to get.</param>
+        /// <param name="value">The value of the object from the collection.</param>
+        /// <returns>true if the collection contains an element with the specified key; otherwise, false.</returns>
```

Comment:
"otherwise, false" does not stand on its own as a sentence. So either make it a sentence (subject, verb) or replace the semicolon with a comma.